www/components/gallery/CodeBox.tsx

```diff
@@ -0,0 +1,43 @@
+import { useState } from "preact/hooks";
+
+import Prism from "https://esm.sh/prismjs@1.27.0";
+import "https://esm.sh/prismjs@1.27.0/components/prism-typescript?no-check";
+
+interface CodeBoxProps {
+  code: string;
+}
+
+export default function CodeBox(props: CodeBoxProps) {
+  const highlighted = Prism.highlight(
+    props.code,
+    Prism.languages.typescript,
+    "typescript",
+  );
+  const onCopy = () => {
+    navigator.clipboard.writeText(props.code);
+    setCopied(true);
+  };
+
+  const [copied, setCopied] = useState(false);
```

Comment:
shouldnt this be in an island?

Reply:
CodeBox.tsx is used in `islands/ComponentGallery.tsx`, so I think it is fine as a component.

Reply:
ah wasnt aware that works; and yea, islands cant import islands, so this is fine as-is.

Reply:
Oh, dependencies of islands can use these hooks. This sounds a tricky situation...